src/main/java/me/rigamortis/seppuku/impl/module/misc/HotBarRefillModule.java

```diff
@@ -0,0 +1,185 @@
+import net.minecraft.client.gui.inventory.GuiInventory;
+import net.minecraft.init.Items;
+import net.minecraft.inventory.ClickType;
+import net.minecraft.item.ItemStack;
+import team.stiff.pomelo.impl.annotated.handler.annotation.Listener;
+
+import java.util.ArrayList;
+
+/**
+ * Automatically refills the players hotbar.
+ *
+ * @author Old Chum
+ * @since 12/7/19
+ */
+public class HotBarRefillModule extends Module {
+    public final Value<Float> delay = new Value<>("Delay", new String[]{"Del"}, "The amount of delay in milliseconds.", 50.0f);
+    public final Value<Integer> refills = new Value<>("NumPerRefill", new String[]{"refill"}, "The number of total slots to be refilled at a time.", 2, 1, 10, 1);
+    public final Value<Boolean> offHand = new Value<>("OffHand", new String[]{"oh", "off", "hand"}, "If the off hand should be refilled.", true);
+
+    private Timer timer = new Timer();
+
+    public HotBarRefillModule() {
+        super("HotBarRefill", new String[]{"Replenish", "Refill", "AutoHotBar", "hbr"}, "NONE", -1, ModuleType.MISC);
+    }
+
+    @Listener
+    public void onWalkingUpdate (EventUpdateWalkingPlayer event) {
+        if (this.timer.passed(this.delay.getValue())) {
+            if (event.getStage() == EventStageable.EventStage.PRE) {
+                Minecraft mc = Minecraft.getMinecraft();
+                
+                if (mc.currentScreen instanceof GuiInventory) {
+                    return;
+                }
+
+                ArrayList<Integer> toRefill = getRefillable(mc.player, getHotbar(mc.player));
+
+                if (!toRefill.isEmpty()) {
+                    for (int i = 0; i < toRefill.size() && i < refills.getValue(); i++) {
+                        refillHotbarSlot(mc, toRefill.get(i));
+                    }
+                }
+            }
+
+            timer.reset();
+        }
+    }
+
+    /**
+     * Gets a list of slots in a hotbar that can be refilled.
+     * If offhand is on the list includes the offhand
+     *
+     * @param hotbar The hotbar
+     * @return A list of slots in a hotbar that can be refilled
+     */
+    private ArrayList<Integer> getRefillable(EntityPlayerSP player, ItemStack[] hotbar) {
+        ArrayList<Integer> ret = new ArrayList<>();
+
+        for (int i = 0; i < hotbar.length; i++) {
+            ItemStack stack = hotbar[i];
+            if (stack.getCount() < stack.getMaxStackSize() && stack.getItem() != Items.AIR) {
+                ret.add(i);
+            }
+        }
+
+        if (offHand.getValue()) {
+            if (player.getHeldItemOffhand().getCount() < player.getHeldItemOffhand().getMaxStackSize()) {
+                ret.add(45);
+            }
+        }
+
+        return ret;
+    }
+
+    /**
+     * Returns a given player's hotbar.
+     *
+     * @param player The player who's hotbar should be returned.
+     * @return The passed player's hotbar as a primitive array.
+     */
+    private ItemStack[] getHotbar (EntityPlayerSP player) {
```

Comment:
This function needs a re-think.  This can be improved dramatically, the array creation isn't the most convincing way to do this.

Reply:
Fixed.